CRM/Contribute/Form/AbstractEditPayment.php

```diff
@@ -213,6 +213,7 @@ public function preProcess() {
     $this->assign('contactID', $this->_contactID);
     CRM_Core_Resources::singleton()->addVars('coreForm', array('contact_id' => (int) $this->_contactID));
     $this->_action = CRM_Utils_Request::retrieve('action', 'String', $this, FALSE, 'add');
+    $this->_mode = empty($this->_mode) ? CRM_Utils_Request::retrieve('mode', 'String', $this) : $this->_mode;
```

Comment:
thanks - some tricky test scenario - slowly hopefully we can work towards the forms bein gmore aligned